### ISSUE TITLE
Installation problem.

### ISSUE THREAD

**Issue by as7a5** [landau-lab/Gotcha#3]
We are using module environment and python is installed in a shared file system (not the default location). at the moment of installation we get the following error:

* installing *source* package ‘Gotcha’ ...
** using staged installation
** R
** byte-compile and prepare package for lazy loading
** help
*** installing help indices
** building package indices
** testing if installed package can be loaded from temporary location
Failed to import the site module
Traceback (most recent call last):
  File "/gpfs/share/apps/python/cpu/3.6.5/lib/python3.6/site-packages/site.py", line 73, in <module>
    __boot()
  File "/gpfs/share/apps/python/cpu/3.6.5/lib/python3.6/site-packages/site.py", line 39, in __boot
    raise ImportError("Couldn't find the real 'site' module")
ImportError: Couldn't find the real 'site' module
ERROR: loading failed
* removing ‘/gpfs/share/apps/R/4.1.2/lib64/R/library/Gotcha’
Warning message:
In i.p(...) :
  installation of package ‘/tmp/RtmpGaJ7Z6/file1594c225541a9/Gotcha_1.0.1.tar.gz’ had non-zero exit status
> 

any help is appreciated.

**Comment by fizzo13** [comment]
When loading the Gotcha package for the first time, it attempts to create a python virtual environment via reticulate.
The full command for creating the environment is:

reticulate::virtualenv_create(envname = "r-reticulate-gotcha",
                                  python = Sys.which("python3"),
                                  version = "python3",
                                  packages = c("pip","wheel"),
                                  pip_version = "pip-22.1.1")

Can you try running this section to see if you can successfully create the environment before installation?

However, regarding the error message "Failed to import the site module", this seems to be related to something being amiss in the python installation since the site.py should be located in the Lib folder (not in "/gpfs/share/apps/python/cpu/3.6.5/lib/python3.6/site-packages/site.py"):

https://docs.python.org/3/library/site.html?module-site

I suggest verifying the python installation for _Sys.which("python3")_ and trying again.

**Comment by as7a5** [comment]
Sorry forgot to mention we are running on RedHat 7.9

I think the code is looking for the default location because on the installation directory there is no “Lib”. 

Also:

***@***.*** ~]$ python
Python 3.6.5 (default, Aug 16 2018, 10:25:11) 
[GCC 4.8.5 20150623 (Red Hat 4.8.5-28)] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> Sys.which("python3")
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
NameError: name 'Sys' is not defined
>>> 

and 

***@***.*** ~]$ R

R version 4.1.2 (2021-11-01) -- "Bird Hippie"
Copyright (C) 2021 The R Foundation for Statistical Computing
Platform: x86_64-pc-linux-gnu (64-bit)

R is free software and comes with ABSOLUTELY NO WARRANTY.
You are welcome to redistribute it under certain conditions.
Type 'license()' or 'licence()' for distribution details.

  Natural language support but running in an English locale

R is a collaborative project with many contributors.
Type 'contributors()' for more information and
'citation()' on how to cite R or R packages in publications.

Type 'demo()' for some demos, 'help()' for on-line help, or
'help.start()' for an HTML browser interface to help.
Type 'q()' to quit R.

> reticulate::virtualenv_create(envname = "r-reticulate-gotcha",
+ python = Sys.which("python3"),
+ version = "python3",
+ packages = c("pip","wheel"),
+ pip_version = "pip-22.1.1")
virtualenv: r-reticulate-gotcha
>

**Comment by fizzo13** [comment]
After running the last section in R, was the environment created? 

You can verify by running in R:
> reticulate::virtualenv_exists("r-reticulate-gotcha")
[1] TRUE

Also, just to double check, did you run _Sys.which("python3")_ within the R session?

**Comment by as7a5** [comment]
yes:
> reticulate::virtualenv_exists("r-reticulate-gotcha")
[1] TRUE

**Comment by fizzo13** [comment]
So run the following within your R session:

### Make sure we are using the correct virtual env:
reticulate::use_virtualenv("r-reticulate-gotcha")

### Install required python modules:
   reticulate::py_install(packages = c("pandas","matplotlib","seaborn","numpy","sklearn"),
                           python_version = "python3",
                           pip = T,
                           envname = "r-reticulate-gotcha")

Basically, we are creating the virtual environment manually. Then re-try the Gotcha installation.

**Comment by as7a5** [comment]
Yes and :

> Sys.which("python3") 
                                        python3 
"/gpfs/share/apps/python/cpu/3.6.5/bin/python3"

**Comment by fizzo13** [comment]
Closing since failed installation is associated with permissions on the server. 
Please check that you have the permissions to install the R package as well as the python modules.
